swapnil/CSS/Theory.txt

```diff
@@ -0,0 +1,216 @@
+   -text-align:Specifies left, right, center, or justified alignment.
+
+3)Boxes
+  -margin-top, margin-bottom, margin-left, margin-right, margin
+   Determine the size of the top, bottom, left, and right margins. Setting
+   margin adjusts all of the margins simultaneously.
+
+  -padding-top,padding-bottom, padding-left, padding-right, padding
+   Adjusts the amount of padding on the top, bottom, left, and right sides of
+   the element. Setting padding adjusts all of them simultaneously.
+
+  -width, height
+   Identifies the width and height of the rectangle that contains the formatted   content. Images should be scaled to the specified size, if necessary.
+
+  -float
+   Identifies an element that should float to the left or right of a flow of
+   text, allowing the text to flow around it.
+
+Q:Using inline styles is considered to be horrible by most web designers.
+  What is so bad about it?
+  -It will messed with HTML.and also peoples who are stuck using older browsers   are having issue with using inline css.
+
+Q:What are the benefits of using a linked stylesheet as compared to using
+  embedded styles in the HTML page itself?
+-Everything is stored within single page.
+-once changed/updated, the changes are reflected on all other pages that
+ reference the stylesheet.
+-easier to maintain larger websites.
+
+Q:The table below contains selector combinatory names in the first column and
+  the way a selector combination is depicted in a style sheet, in the second.
+  However, the values are mismatched. You have to re-create this table with the  correct matches.
+  
+  Combinator Name                   CSS syntax
+  Descendant                        p ~ li {  } 
+  Immediate descendant (child)      p + li {  }
+  Sibling                           P > li {   }
+  Adjacent sibling                  p  li {   }
+
+- Combinator Name                   CSS syntax
+  Descendant                        p  li {   }
+  Immediate descendant (child)      P > li {   }
+  Sibling                           p ~ li {  }
+  Adjacent sibling                  p + li {  }
+
+Q:Many web authors use a reset CSS. What problems might happen if they don’t
+  use it?
+-A CSS Reset (or “Reset CSS”) is a short, often compressed (minified) set of
+ CSS rules that resets the styling of all HTML elements to a consistent
+ baseline.this creates a certain amount of headaches for CSS authors, who can’t work out how to make their websites look the same in every browser.
+
+Q:In which colour will the text “This is the first paragraph.” be printed for
+  the HTML and CSS shown below? Also explain the specificity rule by which that  colour is chosen.
+<html>
+        <head>
+                <style>
+div#section1 span{color:green}
+div p{color:blue}
+p{color:red}
+                </style>
+        </head>
+        <body>
+                <div id="section1">
+                        <p> this is first paragraph </p>
+                </div>
+        </body>
+</html>
+
```

Comment:
bad css code indentation 
use tab (4 spaces)

Reply:
done changes

Reply:
bad indentation from line no. 76 to 80.